contracts/concordium-governance-committee-election/src/lib.rs

```diff
@@ -0,0 +1,314 @@
+pub struct InitParameter {
+    /// The account used to perform administrative functions, such as publishing
+    /// the final result of the election. If this is `None`, the account
+    /// used to instantiate the contract will be used.
+    pub admin_account:        Option<AccountAddress>,
+    /// A list of candidates that voters can vote for in the election.
+    pub candidates:           Vec<Candidate>,
+    /// The list of guardians for the election.
+    pub guardians:            Vec<AccountAddress>,
+    /// The merkle root of the list of eligible voters and their respective
+    /// voting weights.
+    pub eligible_voters:      EligibleVoters,
+    /// A description of the election, e.g. "Concordium GC election, June 2024".
+    pub election_description: String,
+    /// The start time of the election, marking the time from which votes can be
+    /// registered.
+    pub election_start:       Timestamp,
+    /// The end time of the election, marking the time at which votes can no
+    /// longer be registered.
+    pub election_end:         Timestamp,
+}
+
+impl InitParameter {
+    /// Converts the init parameter to [`State`] with a supplied function for
+    /// getting a fallback admin account if none is specified. The function
+    /// consumes the parameter in the process.
+    fn into_state(
+        self,
+        ctx: &InitContext,
+        state_builder: &mut StateBuilder,
+    ) -> Result<State, Error> {
+        let admin_account = self.admin_account.unwrap_or_else(|| ctx.init_origin());
+
+        let config = Config::new_checked(
+            ctx,
+            admin_account,
+            self.candidates,
+            HashSet::from_iter(self.guardians.into_iter()),
+            self.eligible_voters,
+            self.election_description,
+            self.election_start,
+            self.election_end,
+        )?;
+        let state = State {
+            config:          state_builder.new_box(config),
+            election_result: state_builder.new_box(None),
+        };
+        Ok(state)
+    }
+}
+
+/// Init function that creates a new smart contract with an initial [`State`]
+/// derived from the supplied [`InitParameter`]
+#[init(
+    contract = "concordium_governance_committee_election",
+    parameter = "InitParameter",
+    error = "Error"
+)]
+fn init(ctx: &InitContext, state_builder: &mut StateBuilder) -> InitResult<State> {
+    let parameter: InitParameter = ctx.parameter_cursor().get()?;
+    let initial_state = parameter.into_state(ctx, state_builder)?;
+    Ok(initial_state)
+}
+
+/// Temporary until election guard has an encrypted ballot.
+#[derive(Serialize, SchemaType)]
+pub struct Vote {
+    pub candidate_index: u8,
+    pub has_vote:        bool,
+}
+
+/// Temporary until election guard implements an encrypted version of this.
+pub type Ballot = Vec<Vote>;
+
+/// The parameter supplied to the [`register_votes`] entrypoint.
+pub type RegisterVoteParameter = Ballot;
```

Comment:
They are subject to change as we don't know the structure. This is just to get something out that we can interact with from a dApp and server (until election guard implementation of encrypted votes is ready)

Reply:
Ok, but my point is that there won't be a structure. The best approximation will be Vec<u8>.

There won't be anything meaningful to display to the user in the wallet since what they're going to be submitting to the contract will be an encryption of the vote + some proofs.